File: 1872-stone-game-viii/1872-stone-game-viii.java
class Solution {
    public int stoneGameVIII(int[] A) {
        int n = A.length;
        for (int i = 1; i < n; i++)
            A[i] += A[i - 1];
        int ans = A[n - 1];
        for (int i = n - 2; i > 0; i--)
            ans = Math.max(ans, A[i] - ans);
        return ans;
    }
}

File: 1872-stone-game-viii/1872-stone-game-viii.py
class Solution:
    def stoneGameVIII(self, A: List[int]) -> int:
        n = len(A)
        for i in range(1, n):
            A[i] += A[i - 1]
        ans = A[-1]
        for i in range(n - 2, 0, -1):
            ans = max(ans, A[i] - ans)
        return ans